### ISSUE TITLE
Node 8 'tx' endpoint doesn't return the number of confirmations

### ISSUE THREAD

**Issue by osagga** [bitpay/bitcore-node#575]
When using the `tx` endpoint on bitcore-node `v8.0.0`, the returned object has the following format

BTC
```
{
        "_id": "5b635ebfc18edd8496e2f707",
        "chain": "BTC",
        "network": "testnet",
        "txid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915",
        "blockHash": "00000000011a135e5277f5493c52c66829792392632b8b65429cf07ad3c47a6c",
        "blockHeight": 200007,
        "blockTime": "2014-03-01T07:41:25.000Z",
        "blockTimeNormalized": "2014-03-01T07:42:17.051Z",
        "coinbase": false,
        "locktime": 0,
        "size": 225,
        "wallets": [],
        "inputs": [
            {
                "_id": "5b635ebfc18edd8496e2f6ab",
                "chain": "BTC",
                "mintIndex": 1,
                "mintTxid": "b701b0680bc2b6f8b58f13d035a754486ecefd2438e3495cd8f395ab7f5e7ba9",
                "network": "testnet",
                "address": "mqtYvjsUg59XpngBAYjFF51Xauc28a6Ckd",
                "coinbase": false,
                "mintHeight": 200006,
                "script": "dqkUccbnHilMfAYMHYBz+Gr2E0bGPk6IrA==",
                "spentHeight": 200007,
                "value": 34540000,
                "wallets": [],
                "spentTxid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915"
            },
            {
                "_id": "5b635ee3c18edd8496e4ae1d",
                "chain": "BCH",
                "mintIndex": 1,
                "mintTxid": "b701b0680bc2b6f8b58f13d035a754486ecefd2438e3495cd8f395ab7f5e7ba9",
                "network": "testnet",
                "address": "bchtest:qpcudec799x8cpsvrkq887r27cf5d337fcl2x3rvq4",
                "coinbase": false,
                "mintHeight": 200006,
                "script": "dqkUccbnHilMfAYMHYBz+Gr2E0bGPk6IrA==",
                "spentHeight": 200007,
                "value": 34540000,
                "wallets": [],
                "spentTxid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915"
            }
        ],
        "outputs": [
            {
                "_id": "5b635ebfc18edd8496e2f6e4",
                "chain": "BTC",
                "mintIndex": 0,
                "mintTxid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915",
                "network": "testnet",
                "address": "muZWTYvwGs3nMqS72hvMsJ2zERkvENpi8y",
                "coinbase": false,
                "mintHeight": 200007,
                "script": "dqkUmg1xHG+j2nExDPZu1i/2uOQ0eaKIrA==",
                "spentHeight": 200007,
                "value": 29530000,
                "wallets": [],
                "spentTxid": "2dbf3b306984ec6ca69dc06d3fdbdc1b36c3ce5c467c028cfad64c1c99e3ff7a"
            },
            {
                "_id": "5b635ebfc18edd8496e2f6e5",
                "chain": "BTC",
                "mintIndex": 1,
                "mintTxid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915",
                "network": "testnet",
                "address": "2NDcM3CGUTwqFL7y8BSBJTYJ9kToeXawkUF",
                "coinbase": false,
                "mintHeight": 200007,
                "script": "qRTfYSLV7bdpe0ZvcgGoYrT9Z+oqwYc=",
                "spentHeight": 200577,
                "value": 5000000,
                "wallets": [],
                "spentTxid": "3cbf42f789dee14c43f2eb359625554c2f296c34750ed686b98422b1b2e2a035"
            },
            {
                "_id": "5b635ee3c18edd8496e4ae66",
                "chain": "BCH",
                "mintIndex": 0,
                "mintTxid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915",
                "network": "testnet",
                "address": "bchtest:qzdq6ugud73a5uf3pnmxa43076uwgdre5gyd7tjela",
                "coinbase": false,
                "mintHeight": 200007,
                "script": "dqkUmg1xHG+j2nExDPZu1i/2uOQ0eaKIrA==",
                "spentHeight": 200007,
                "value": 29530000,
                "wallets": [],
                "spentTxid": "2dbf3b306984ec6ca69dc06d3fdbdc1b36c3ce5c467c028cfad64c1c99e3ff7a"
            },
            {
                "_id": "5b635ee3c18edd8496e4ae67",
                "chain": "BCH",
                "mintIndex": 1,
                "mintTxid": "3e81723d069b12983b2ef694c9782d32fca26cc978de744acbc32c3d3496e915",
                "network": "testnet",
                "address": "bchtest:pr0kzgk4akmkj76xdaeqr2rzkn7k0632cyuzwmtahu",
                "coinbase": false,
                "mintHeight": 200007,
                "script": "qRTfYSLV7bdpe0ZvcgGoYrT9Z+oqwYc=",
                "spentHeight": 200577,
                "value": 5000000,
                "wallets": [],
                "spentTxid": "3cbf42f789dee14c43f2eb359625554c2f296c34750ed686b98422b1b2e2a035"
            }
        ]
    }
```

BCH
```
[
    {
        "_id": "5b638eabc18edd84960219a1",
        "chain": "BCH",
        "network": "testnet",
        "txid": "69c2ce001ea316bee14423982d2f07e831cf4ab424c75bb29448410e1bdd4078",
        "blockHash": "000000007306c5ff2e6c2bf09260c94d3c05134cfcfe3a2d84151c58d02db642",
        "blockHeight": 277199,
        "blockTime": "2014-08-25T00:48:58.000Z",
        "blockTimeNormalized": "2014-08-25T00:48:58.000Z",
        "coinbase": true,
        "locktime": 0,
        "size": 109,
        "wallets": [],
        "inputs": [],
        "outputs": [
            {
                "_id": "5b638656c18edd8496f8362d",
                "chain": "BTC",
                "mintIndex": 0,
                "mintTxid": "69c2ce001ea316bee14423982d2f07e831cf4ab424c75bb29448410e1bdd4078",
                "network": "testnet",
                "address": "mfvXaxf3xSCpVnDafuZuUPePZ6TVQyfdHC",
                "coinbase": true,
                "mintHeight": 277199,
                "script": "IQKYDcOKCdssfuBYTtzY0wCHQ8rbY3TzP7bMuW62eEA/Tqw=",
                "spentHeight": 277660,
                "value": 2500040030,
                "wallets": [],
                "spentTxid": "873f002a9ae9085d69bf3dd62e7cf467f327de58c52a814cf8dc76d7ba2bceb2"
            },
            {
                "_id": "5b638eaac18edd8496021949",
                "chain": "BCH",
                "mintIndex": 0,
                "mintTxid": "69c2ce001ea316bee14423982d2f07e831cf4ab424c75bb29448410e1bdd4078",
                "network": "testnet",
                "address": "bchtest:qqz82kuzfj28lrhjnkah40ym237h73ajasgyhh9gxh",
                "coinbase": true,
                "mintHeight": 277199,
                "script": "IQKYDcOKCdssfuBYTtzY0wCHQ8rbY3TzP7bMuW62eEA/Tqw=",
                "spentHeight": -2,
                "value": 2500040030,
                "wallets": []
            }
        ]
    }
]
```

LTC
```
{
        "_id": "5b635cc4c18edd8496c6ea91",
        "chain": "LTC",
        "network": "testnet",
        "txid": "91d5a6be3832121b71db67e2ddcd873479983ea9ae0b614f3af7ab1d760b89a8",
        "blockHash": "736dacad05bf676aa3f8cef7eb760657ea44d270c2859e173af5339f925139c2",
        "blockHeight": 200009,
        "blockTime": "2017-09-24T04:15:02.000Z",
        "blockTimeNormalized": "2017-09-24T04:15:02.000Z",
        "coinbase": false,
        "locktime": 200008,
        "size": 487,
        "wallets": [],
        "inputs": [
            {
                "_id": "5b635663c18edd849678807e",
                "chain": "LTC",
                "mintIndex": 0,
                "mintTxid": "da6694c3ac674e70588b49d187036ef52531341e1bbfce8eb88867d7a4e19939",
                "network": "testnet",
                "address": "n3znnvqx498hUBgXcX9PX33MEs9yPMAsXa",
                "coinbase": true,
                "mintHeight": 156660,
                "script": "dqkU9pZ7b9sK1rHYy7l/zgep1W/FYQ2IrA==",
                "spentHeight": 200009,
                "value": 5000000000,
                "wallets": [],
                "spentTxid": "91d5a6be3832121b71db67e2ddcd873479983ea9ae0b614f3af7ab1d760b89a8"
            },
            {
                "_id": "5b63572bc18edd849681138d",
                "chain": "LTC",
                "mintIndex": 0,
                "mintTxid": "4a9201ea1159f460c3608f897cb2e6972f4658ef2bc56e6fe9dc5bcb4d80c6a4",
                "network": "testnet",
                "address": "n3znnvqx498hUBgXcX9PX33MEs9yPMAsXa",
                "coinbase": true,
                "mintHeight": 160840,
                "script": "dqkU9pZ7b9sK1rHYy7l/zgep1W/FYQ2IrA==",
                "spentHeight": 200009,
                "value": 5000000000,
                "wallets": [],
                "spentTxid": "91d5a6be3832121b71db67e2ddcd873479983ea9ae0b614f3af7ab1d760b89a8"
            },
            {
                "_id": "5b635cc4c18edd8496c6e96a",
                "chain": "LTC",
                "mintIndex": 1,
                "mintTxid": "99b1dc39104c12143d5dc78b464993fdc2c8390031621cfb787da132c10e9fc2",
                "network": "testnet",
                "address": "n4PnqyoXRzmQJN4XdW1zFBpw8vsFHwT2dc",
                "coinbase": false,
                "mintHeight": 200006,
                "script": "dqkU+vAX3HjAwsC+sTxxOi/oxIuV53uIrA==",
                "spentHeight": 200009,
                "value": 10000000,
                "wallets": [],
                "spentTxid": "91d5a6be3832121b71db67e2ddcd873479983ea9ae0b614f3af7ab1d760b89a8"
            }
        ],
        "outputs": [
            {
                "_id": "5b635cc4c18edd8496c6ea6e",
                "chain": "LTC",
                "mintIndex": 0,
                "mintTxid": "91d5a6be3832121b71db67e2ddcd873479983ea9ae0b614f3af7ab1d760b89a8",
                "network": "testnet",
                "address": "mh4EbR5L5Vd4ttByET5VGTb8ThtANnrxFh",
                "coinbase": false,
                "mintHeight": 200009,
                "script": "dqkUEOKgaTTXi/MRnCSjRh9dq4hEDMuIrA==",
                "spentHeight": -2,
                "value": 10000000000,
                "wallets": []
            }
        ]
    }
```

The transaction object does not contain the number of confirmations the transaction has received. The number of confirmations is really important and critical to a block explorer.

**Comment by micahriggan** [comment]
The confirmations is a derived value from localChainTip.height-tx.height

I believe we may be doing that in insight.

**Comment by osagga** [comment]
Yeah that's true, but I think it would be faster and more convenient if the `tx` API endpoint returns the confirmations. 

If the API user needs to derive the confirmations, the user will need to make two API calls per transaction (one to get the current chain tip, and another to get the `tx.height`) to calculate the confirmations, instead of only one if the API provides it.

All the node has to do is to locally compute the confirmations when a transaction is requested by the API (or multiple transactions.) Hence, not increasing the storage, but slightly impacting the response time (not by a lot compared to the user making two API calls to compute the same value).

**Comment by osagga** [comment]
Fixed, check bitpay/bitcore#1692
